Documentation/Open Location Code API Specification.md

```diff
@@ -0,0 +1,78 @@
+
+* Version 1.1.0 / 2021-06-04 / [William Entriken](https://github.com/fulldecent/)
+  * Update to match nomenclature in [Plus Codes specification](./Plus%20Codes%20Specification.md)
+  * Separate REQUIRED and OPTIONAL public methods
+  * Specify the number of significant digits that a short code Plus Code may omit
+
+- Version 1.0.0 / 2014-10-27 / [Doug Rinckes](https://github.com/drinckes), Google / Philipp Bunge, Google
+  - Initial public release
+
+## REQUIRED public methods
+
+An implementation of the Open Location Code API SHALL implement these REQUIRED methods:
+
+| Method name          | Input(s)                        | Output(s)                                                    |
+| -------------------- | ------------------------------- | ------------------------------------------------------------ |
+| `isFull`             | Plus Code string                | True if this is a long code Plus Code, false otherwise       |
+| `encode`             | Latitude, longitude             | The Plus Code with code length 10 with area that contains the specified coordinates |
+| `decode`             | Full code Plus Code string with length 10 | The southern parallel and western meridian for the Plus Code area |
+
+Implementations are RECOMMENDED to use the method names above.
+
+Note: it is possible to implement `isFull` using a Perl Compatible Regular Expression, see the Plus Code specification.
+
+## OPTIONAL public methods
+
+An implementation MAY implement these methods.
+
+| Method name        | Input(s)                                                     | Output(s)                                                    |
+| ------------------ | ------------------------------------------------------------ | ------------------------------------------------------------ |
+| `isValid`          | Plus Code string                                             | True if this is a Plus Code (full code or short code), false otherwise |
+| `isShort`          | Plus Code string                                             | True if this is a short code Plus Code, false otherwise      |
+| `encode`           | Latitude, longitude, code length                             | The full code Plus Code with specified code length with area that contains the specified coordinates |
+| `decode`           | Full code Plus Code string                                   | The southern parallel and western meridian for the Plus Code area |
+| `shorten`          | Full code Plus Code, reference location latitude and longitude | The short code Plus Code representing the input where all significant digits that are allowed to be omitted are omitted; or if no digits are allowed to be omitted then the input is returned |
+| `recoverNearest`   | Short code Plus Code, reference location latitude and longitude | The unique full code Plus Code nearest (latitudinal distance + longitudinal distance) to the reference location |
```

Comment:
In the output column, "(latitudinal distance + longitudinal distance)" seems to suggest using rectilinear distance instead of euclidean distance as the metric to determine the nearest location. Suggestion: mention the actual metric by name, or remove quoted text if an exact metric does not need to be mandated.